tools/clang/test/CodeGenSPIRV/vk.subpass-input.unused.hlsl

```diff
@@ -0,0 +1,23 @@
+// Test that the we can correctly compile the compute shader when the
+// subpass input is used in  PsSubpassTest, but not in the compute shader.
+
+// RUN: %dxc -T cs_6_0 -E CsTest
+
+// The variable is declared, but should not be used.
+// CHECK: %subInput = OpVariable
```

Comment:
Are we sure this wouldn't break some drivers?
Seems like this would go against the SPIR-V spec, as it states of OpTypeImage has SubpassData, the execution type must be fragment.
This should get DCOd, so we must request the legalization pass no? (And have a test making sure we don't emit the variable)

Reply:
Fixed. I just avoided added the declaration.